crates/sozo/src/commands/dev.rs

```diff
@@ -0,0 +1,265 @@
+            RecursiveMode::Recursive,
+        )?;
+        let name = self.name.clone();
+        let mut previous_manifest: Option<Manifest> = Option::None;
+        let mut result = build(&mut context);
+
+        let mut env_metadata =
+            dojo_metadata_from_workspace(&context.ws).and_then(|inner| inner.env().cloned());
+        let Some((mut world_address, account)) = context
+            .ws
+            .config()
+            .tokio_handle()
+            .block_on(migration::setup_env(
+                self.account,
+                self.starknet,
+                self.world,
+                env_metadata.as_ref(),
+                config,
+                name.as_ref(),
+            ))
+            .ok()
+        else {
+            return Err(anyhow!("Failed to setup environment"));
+        };
+
+        match context.ws.config().tokio_handle().block_on(migrate(
+            world_address,
+            &account,
+            name.clone(),
+            &context.ws,
+            previous_manifest.clone(),
+        )) {
+            Ok((manifest, address)) => {
+                previous_manifest = Some(manifest);
+                world_address = address;
+            }
+            Err(error) => {
+                log::error!("Error: {error:?}");
+            }
+        }
+        loop {
+            let mut action = DevAction::None;
+
+            match rx.recv() {
+                Ok(events) => {
+                    if events.is_ok() {
+                        events.unwrap().iter().for_each(|event| {
+                            action = handle_event(event);
+                            match &action {
+                                DevAction::None => {}
+                                DevAction::Build(path) => {
+                                    context.ws.config().ui().print(format!(
+                                        "📦 Need to rebuild {}",
+                                        path.clone().as_path().to_str().unwrap(),
+                                    ));
+                                    let db = &mut context.db;
+                                    let file = FileId::new(db, path.clone());
+                                    PrivRawFileContentQuery
+                                        .in_db_mut(db.as_files_group_mut())
+                                        .invalidate(&file);
+                                    db.override_file_content(file, None);
+                                }
+                                DevAction::Reload => {
+                                    context
+                                        .ws
+                                        .config()
+                                        .ui()
+                                        .print("Reloading project");
+                                    context = load_context(config).unwrap();
+                                    env_metadata = dojo_metadata_from_workspace(&context.ws)
```

Comment:
is this `env_metadata` value used? are we relying on a side effect here?

Reply:
I don't know yet how we should handle environment related modifications in project.

From my understanding, environment could change when at least `scarb.toml` is modified but I don't have the developer experience if this is a really needed when developing with dojo.

Reply:
I think in the case on a change to scarb.toml, we can ask the user to restart dev mode, some there are changes there would be hard to not reload, like configuration data